crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -553,12 +578,12 @@ impl<T: Component> WorldQuery for &T {
 #[doc(hidden)]
 pub struct ReadState<T> {
     component_id: ComponentId,
-    marker: PhantomData<T>,
+    marker: PhantomData<fn() -> T>,
```

Comment:
This aligns with my mental model of how this should be modelled, but this is subtle enough that it deserves a comment.

Reply:
Hmm, I don't really think it's needed. This is very much the standard way to express 'this represents something which can produce a `T`'.

It's no more strange than the initial use of `PhantomData` imo - maybe we should have a comment that we need to be able to 'reconstruct' the original `T`, so store it here. But we don't need to comment specifically on it being `fn()->T`